### ISSUE TITLE
[MSKINS-235] wrong META-INF/LICENSE and META-INF/NOTICE in release jar

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-fluido-skin#240]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** opened **[MSKINS-235](https://issues.apache.org/jira/browse/MSKINS-235?redirect=false)** and commented

seen as testing reproducible builds: https://github.com/jvm-repo-rebuild/reproducible-central/blob/master/content/org/apache/maven/skins/fluido/README.md

reference jar has content that seems to be generated from remote-resources with Jar Resource Bundle https://maven.apache.org/apache-resource-bundles/jar/

but rebuild has content configured in pom.xml to copy content from Git: https://github.com/apache/maven-fluido-skin/
 

---

**Affects:** fluido-2.0.0-M5, fluido-2.0.0-M6, fluido-2.0.0-M7

**Issue Links:**
- [MRESOURCES-278](https://issues.apache.org/jira/browse/MRESOURCES-278) Regression - Changed copy behaviour if file already exists
 (_**"is caused by"**_)

**Remote Links:**
- [GitHub Pull Request #54
](https://github.com/apache/maven-fluido-skin/pull/54)

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

this release used maven-remote-resources 1.7.0, not latest 3.1.0: we need to check if simply upgrading to 3.1.0 solves the issue
we need w Windows box to test...

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

ok, the copy does not happen in maven-remote-resources-plugin but in maven-resources-plugin:
upgrading maven-resources-plugin from 3.3.0 to 3.3.1 gives more explicit output:

```
[INFO] --- resources:3.3.1:resources (default-resources) @ maven-fluido-skin ---
[INFO] Copying 2 resources from  to target/classes/META-INF
[INFO] Copying 33 resources from src/main/resources to target/classes
[INFO] Copying 3 resources from src/main/resources to target/classes
[INFO] Copying 1 resource from src/main/resources to target/classes
[INFO] Copying 2 resources from target/maven-fluido-skin-2.0.0-M8-SNAPSHOT to target/classes
[INFO] Copying 3 resources from target/maven-shared-archive-resources to target/classes
```

the first and the last copy lines are the interesting ones:
- first: "Copying 2 resources from  to target/classes/META-INF" copies the content root in Git = the handwritten LICENSE and NOTICE that we expect will be kept
- last: "Copying 3 resources from target/maven-shared-archive-resources to target/classes" = the generated content from apache-resource-bundles

I tested on a Linux box and on a Windows box
Surprisingly on Linux, the overwrite from the second copy does not happen
but on Windows, the second copy overwrites

The Windows behaviour looks reasonable, even if it's not the result we want (we want to keep the hand-written files)

IMHO:
1. there is a bug on Linux on maven-resources-plugin
2. the way maven-fluido-skin expects no force hand-written content looks wrong, because it's overwritten by generated content

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

using maven-resources-plugin 3.2.0 (that was used in 2.0.0-M3) solves the issue: plugin changes behaviour on Windows when going from 3.2.0 to 3.3.0

one workaround for now is just to force plugin release, before fixing it

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [85fa18cd3525ac961bf71e7aea0badd3fd08feaf](https://gitbox.apache.org/repos/asf?p=maven-fluido-skin.git;a=commit;h=85fa18cd3525ac961bf71e7aea0badd3fd08feaf).
